### PR TITLE
embed recaptcha execute and reset methods

### PR DESCRIPTION
**Pull request by dennis90**
## Changelog
- Embed `window.grecaptcha` method inside the hook
- Updated tests and documentations
- Updated hook jsdoc return type

## Motivation
- This will make easier to create typescript mappings for this project and avoid the need to declare new `window` methods on projects that use this lib

### CODE REVIEW

File: src/useRecaptcha.test.js [hunk 1]
```diff
@@ -23,7 +23,7 @@ function createReCaptchaMock() {
 
 window.grecaptcha = createReCaptchaMock();
 
-const TestVisibleRecaptcha = () => {
+const TestInvisibleRecaptcha = () => {
```
Comment (tomliangg) [inline src/useRecaptcha.test.js:26]: nice! Thanks for correcting the typos

File: src/useRecaptcha.js [hunk 2]
```diff
@@ -29,7 +29,7 @@ export function useRecaptcha({
   errorCallback,
 }) {
   // https://stackoverflow.com/questions/45240833/what-are-the-benefits-of-explicitly-rendering-recaptcha-widget-as-opposed-to-aut
-  // explicit rendering is needed because the view template of applicatio may not be loaded yet when recaptcha is loaded
+  // explicit rendering is needed because the view template of application may not be loaded yet when recaptcha is loaded
```
Comment (tomliangg) [inline src/useRecaptcha.js:32]: 👍

File: src/localDev/InvisibleCaptcha.jsx [hunk 3]
```diff
@@ -19,10 +19,8 @@ export default function InvisibleCaptcha() {
 
   const executeCaptcha = (e) => {
     e.preventDefault();
-    if (recaptchaWidget !== null) {
-      window.grecaptcha.reset(recaptchaWidget);
-      window.grecaptcha.execute(recaptchaWidget);
-    }
+    execute();
+    reset();
```
Comment (tomliangg) [inline src/localDev/InvisibleCaptcha.jsx:23]: `reset()` was called before `execute()`. We shall preserve that order.

**Review by tomliangg — COMMENTED** [review]
Your PR looks great! 
I only have one minor holdback.

**Comment by dennis90** [comment]
> Your PR looks great! I only have one minor holdback.

thanks @tomliangg, addressed these problems!

**Review by tomliangg — APPROVED** [review]

**Comment by tomliangg** [comment]
Published 1.0.5;
Release notes: https://github.com/tomliangg/react-hook-recaptcha/releases/tag/1.0.5

**Pull request merged**